index.bs

```diff
@@ -1897,6 +1924,17 @@ object |transport|, and a |sendOrder|, run these steps.
 
 </div>
 
+# `WebTransportAtomicWriter` Interface #  {#web-transport-atomic-writer-interface}
+
+{{WebTransportAtomicWriter}} is a specialized version of {{WritableStreamDefaultWriter}}.
+
+<pre class="idl">
+[Exposed=*]
+interface WebTransportAtomicWriter : WritableStreamDefaultWriter {
+  Promise&lt;bool&gt; write(optional any chunk);
```

Comment:
Adding a `Promise<unsigned long long> getAvailable()` function might help in the recovery scenario.  That function would come with a pretty big caveat: the value you get isn't guaranteed to be the value that is applied when you subsequently write.  But it might help guide decision-making during recovery in the general case, provided that the browser and app are fast in processing and acting upon the result.

Reply:
Would that number match the number of bytes that got sent? If so, we could instead resolve `write()` with the number of bytes that got sent... maybe?

Reply:
That's sort of how the API works in Cloudflare quiche, you can ask to send N bytes and we'll return X (between 1 and N) - https://docs.quic.tech/quiche/struct.Connection.html#method.stream_send

Reply:
Returning how many bytes were written makes sense for a non-atomic API - we do the same in our stack - but it doesn't make sense here.  This is expressly an all-or-nothing option.  The idea was to provide a query that reports many bytes would/could have been written if a write were attempted at the time of the query.

Reply:
@martinthomson OK how about we redefine the number returned from `write` to be the number of bytes (or less) you should try instead? Simply to avoid a separate `getAvailable()` method.

Reply:
Won't that just be confusing?  I know that it streamlines code, but it is still a bit weird.  You write N bytes, and get back M == N to indicate that it worked and M < N to indicate that it didn't?  I don't see why we would want to save the extra call when that is far easier to comprehend.

Reply:
I've reverted back to a boolean to resolve this. Let's discuss `Promise<unsigned long long> getAvailable()` in a new issue if needed.